src/issuance.rs

```diff
@@ -367,17 +410,29 @@ impl IssueBundle<Unauthorized> {
             Some(action) => {
                 // Append to an existing IssueAction.
                 action.notes.push(note);
+                if first_issuance {
+                    action.notes.push(reference_note);
+                }
             }
             None => {
                 // Insert a new IssueAction.
+                let notes = if first_issuance {
+                    vec![reference_note, note]
+                } else {
+                    vec![note]
+                };
                 self.actions.push(IssueAction {
                     asset_desc: Vec::from(asset_desc),
-                    notes: vec![note],
+                    notes,
                     finalize: false,
                 });
             }
         };
 
+        if first_issuance {
+            self.reference_notes.insert(asset, reference_note);
+        }
```

Comment:
This looks overly complicated. Instead of using two `if`s inside the match revert to previous situation and 
```rust
if first_issuance {
  // find the relevant action and insert ref note. Action is guaranteed to exist. Can add assert if not sure.
  
  self.reference_notes.insert(asset, reference_note);
}

```

Reply:
I simplified it with only one "if first_issuance"